lib/sqlalchemy_ingres/base.py

```diff
@@ -221,14 +219,23 @@ def get_column_specification(self, column, **kwargs):
         if str(column.type) == "VARCHAR":
             if column.type.length is None:   # If no length is provided for VARCHAR column
                 column.type.length = 255     # Set length to 255
-        
-        colspec = (
-            self.preparer.format_column(column)
-            + " "
-            + self.dialect.type_compiler_instance.process(
-                column.type, type_expression=column
+
+        if (sqlalchemy_version_tuple >= (2,0)):
+            colspec = (
+                self.preparer.format_column(column)
+                + " "
+                + self.dialect.type_compiler_instance.process(
+                    column.type, type_expression=column
+                )
+            )
+        else:
```

Comment:
tempted to add a comment here to show intent is 1.x code (I don't think that this is worth an actual conditional check here).

Reply:
> The SQLAlchemy repository also contains a large number of "other" tests (https://github.com/sqlalchemy/sqlalchemy/tree/main/test) that are separate from the dialect compliance suite. This "other" test suite has around 14K tests in SQLAlchemy 1.4.51 and 18K tests in 2.0.27. Ref PR #35
> 
> I discovered the dialect compliance suite in April 2024. Prior to that time, I had always run the "other" test suite.

Please open new ticket/PR to get documentation updated for this. This is not easily discoverable as you found out (and I'm only finding out today), nice work figuring this out!

Reply:
Issue [II-16362](https://actian.atlassian.net/browse/II-16362) opened to update Ingres connector readme with details on the SQLAlchemy test suites.